### ISSUE TITLE
pyarmor.helper.repack script keeps the original, unobfuscated code in the executable generated by PyInstaller on macOS

### ISSUE THREAD

**Issue by ntamas** [dashingsoft/pyarmor#414]
It looks like `pyarmor.helper.repack` has a bug when re-packing PyInstaller-generated executables on macOS. This is the code that the script uses on macOS to replace the unobfuscated archive with the new, obfuscated archive at the end of the executable:

```
with open(obfname, 'ab') as outf:
    # Keep bootloader
    outf.seek(offset, os.SEEK_SET)

    # write the patched archive
   with open(pkgname, 'rb') as infh:
        shutil.copyfileobj(infh, outf, length=64*1024)
```

However, since the file is opened in `ab` mode, the seek command has no effect and the new archive is _appended_ to the executable instead of _replacing_ the old one. This can be confirmed by comparing the file size of the original and the new executable: the difference is exactly the length of the new, patched archive.

**Comment by jondy** [comment]
Thanks, it's really an issue, fixed by this commit 4da7f8a8ae7ca2629baed8196260bf7b5cdb35bf

**Comment by ntamas** [comment]
Thanks a lot for the quick fix!
